pkg/asset/installconfig/gcp/dns.go

```diff
@@ -0,0 +1,50 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	dns "google.golang.org/api/dns/v1"
+	"google.golang.org/api/option"
+)
+
+// GetPublicZone returns a DNS managed zone from the provided project which matches the baseDomain
+// If multiple zones match the basedomain, it uses the last public zone in the list as provided by the GCP API.
+func GetPublicZone(ctx context.Context, project, baseDomain string) (managedZone *dns.ManagedZone, err error) {
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
```

Comment:
you use the same context with a timeout of 1 minute to page through all managed dns zones. Is there any concern with high latency connections and many zones?

Reply:
we are filtering on DNS Zones matching on `basedomain`
https://github.com/openshift/installer/pull/1944/commits/2b7bd18d8efcc2b256eaa100dc42d40b47f45932#diff-4475e438b327663f0625c6bfef1d2dd4R33

And there shouldn't be a whole lot of those...